src/event-stream/event-server.ts

```diff
@@ -686,7 +699,14 @@ function createMessageProcessorQueue(): EventMessageHandler {
     },
     handleBurnBlock: (msg: CoreNodeBurnBlockMessage, db: PgWriteStore) => {
       return processorQueue
-        .add(() => observeEvent('burn_block', () => handleBurnBlockMessage(msg, db)))
+        .add(async () => {
+          await observeEvent('burn_block', () => handleBurnBlockMessage(msg, db));
+          if (metrics) {
+            metrics.blocksInPreviousBurnBlock.set(
+              await db.getStacksBlockCountAtBurnBlock(msg.burn_block_height - 1)
+            );
+          }
+        })
```

Comment:
Should we have this in the event-ingestion path? It could be added to the `async collect() { ... }` function of the `metrics.blocksInPreviousBurnBlock` instance so it's only queried when prom metrics are fetched.

Reply:
Great idea, done